tg_bot/modules/users.py

```diff
@@ -65,6 +69,133 @@ def broadcast(bot: Bot, update: Update):
+        message.reply_text("Incorrect number of arguments. Please use `/restrict chat_id`.",
+                           parse_mode=ParseMode.MARKDOWN)
+
+    chat_id = args[0]
+
+    # Check if chat_id is valid
+    if len(args) == 1 and not chat_id.startswith('-'):
+        message.reply_text("Invalid chat id! Make sure you include the '-' sign in the chat id.")
+
+    # Check if chat_id is in bot database
+    all_chats = sql.get_all_chats() or []
+    if len(args) == 1 and chat_id.startswith('-') and not any(chat.chat_id == chat_id for chat in all_chats):
+        message.reply_text("I can't seem to find the chat in my database. "
+                           "Use /chatlist to obtain a list of chats in my database.")
+
+    if len(args) == 1 and chat_id.startswith('-') and any(chat.chat_id == chat_id for chat in all_chats):
+        chat_id_correct = True
+
+    chat_restricted = sql.get_restriction(chat_id)
+    if chat_id_correct and not chat_restricted:
+        try:
+            chat_title = html.escape(bot.get_chat(chat_id).title)
+        except BadRequest as excp:
+            if excp.message == "Chat not found":
+                message.reply_text("Looks like I'm no longer a part of that chat!")
+                return
+            else:
+                raise
+
+        sudo_users_list = "<b>My Admins:</b>"
+        for user in SUDO_USERS:
+            name = mention_html(user, bot.get_chat(user).first_name)
+            sudo_users_list += "\n - {}".format(name)
+
+        bot.send_message(chat_id = chat_id,
+                         text = "I have been restricted by my admins from this chat. "
+                                "Request any of my admins to add me to this chat.\n\n"
+                                "{admins_list}".format(admins_list = sudo_users_list), parse_mode=ParseMode.HTML)
+
+        bot.leave_chat(chat_id)
+
+        sql.set_restriction(chat_id, chat_title, restricted=True)
+
+        message.reply_text("Successfully left chat <b>{}</b>!".format(chat_title), parse_mode=ParseMode.HTML)
+            
+        # Report to sudo users
+        restrictor = update.effective_user  # type: Optional[User]
+        send_to_list(bot, SUDO_USERS, "{} has restricted me from being added to the chat <b>{}</b>."
+                     .format(mention_html(restrictor.id, restrictor.first_name), chat_title), html=True)
+                    
+    elif chat_id_correct and chat_restricted:
+        message.reply_text("I'm already restricted from that chat!")
+                
+
+@run_async
+def new_member(bot: Bot, update: Update): # Leave group when added to restricted group
+    chat = update.effective_chat  # type: Optional[Chat]
+    new_members = update.effective_message.new_chat_members  # type: Optional[User]
+    user = update.effective_user  # type: Optional[User]
+
+    if sql.get_restriction(chat.id):
+        if not user.id in SUDO_USERS:
+            if any(new_mem.id == bot.id for new_mem in new_members):
```

Comment:
Youre doing this if statement twice for each branch of the SUDO_USERS check - consider moving it out, this is duplicate code